external_parser/unit_tests/test_metrics.cc

```diff
@@ -8,20 +8,18 @@ using namespace rapidjson;
 void should_match_expected_metrics(std::string infile_name,
   std::map<std::string, int> expected_metrics) {
   std::string infile = get_test_files_location() + infile_name;
-  std::string outfile = get_test_files_location() + "metrics_spec.json";
+  std::string outfile = get_test_files_location() + "/test_outputs/metrics_spec.json";
+
+  remove(outfile.c_str());
 
   auto vw = VW::initialize("--cb_explore_adf -d " + infile +
                            " --binary_parser --quiet --extra_metrics " + outfile,
                            nullptr, false, nullptr, nullptr);
 
-  v_array<example *> examples = v_init<example*>();
-  examples.push_back(&VW::get_unused_example(vw));
```

Comment:
let's make sure of replacing all of these with generic_driver in case there's more